subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidCachingSmokeTest.groovy

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+import org.eclipse.jgit.api.Git
+
+class AndroidCachingSmokeTest extends AbstractSmokeTest {
+
+    def "can cache Santa Tracker Android application"() {
+        def testRepoUri = "https://github.com/gradle/android-relocation-test"
+        def testRepoBranch = "master"
+
+        def projectDir = testProjectDir.root
+
+        println "Cloning $testRepoUri branch $testRepoBranch..."
+
+        def clone = Git.cloneRepository()
+        clone.URI = testRepoUri
+        clone.branch = testRepoBranch
+        clone.directory = projectDir
+        clone.cloneSubmodules = true
+        clone.call()
+
+        // TODO Log branch / commit ID used in test
+
+        expect:
+        runner(
+                "check",
+                "-Dorg.gradle.android.test.gradle-installation=" + System.getProperty('integTest.gradleHomeDir'),
```

Comment:
I guess we cannot use `IntegrationTestBuildContext` in smoke tests, right?

Reply:
The super-class also uses this approach. I could extract it into a method, though.

Reply:
It is actually available, fixed.